### PR TITLE
Feature ZK-3960: A rating component

### PR DESCRIPTION
**Pull request by wenninghsu**

### CODE REVIEW

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

File: zktest/src/archive/test2/F86-ZK-3960.zul [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                <button onClick='r2.setRating(4)' label="7. set rating to 4"/>
+                <button onClick='r2.setOrient("horizontal".equals(r2.getOrient()) ? "vertical" : "horizontal")' label="8. toggle orient"/>
+                <rating id="r2" orient="vertical" rating="@bind(vm.rating2)"/>
+                <div>Rated: <label value="@bind(vm.rating2)"/></div>
+            </vlayout>
+        </window>
+        <separator width="50px"/>
+        <window border="normal">
+            <vlayout>
+                <button onClick="disable.setDisabled(disable.isDisabled() ? false : true)" label="9. toggle disabled"/>
```
Comment (scribetw) [inline zktest/src/archive/test2/F86-ZK-3960.zul]: `disable.setDisabled(!disable.isDisabled())`

File: zktest/src/archive/test2/F86-ZK-3960.zul [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    <hlayout>
+        <window border="normal" title="Font awesome">
+            <vlayout>
+                <button onClick='r1.setSymbol("z-icon-flag")' label="1. set to Font awesome z-icon-flag"/>
+                <button onClick='r1.setSymbol("U+0260E")' label="2. set to U+0260E &#9742;"/>
+                <button onClick='r1.setMax(10)' label="3. set size 10"/>
+                <button onClick='r1.setMax(5)' label="4. set size 5"/>
+                <rating id="r1" symbol="z-icon-home" rating="@bind(vm.rating1)" cancelable="false"/>
+                <div>Rated: <label value="@bind(vm.rating1)"/></div>
```
Comment (scribetw) [inline zktest/src/archive/test2/F86-ZK-3960.zul]: label just `@load` is enough

File: zktest/src/archive/test2/F86-ZK-3960.zul [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            </vlayout>
+        </window>
+        <separator width="50px"/>
+        <window border="normal" title="default star unicode">
+            <vlayout>
+                <button onClick='r2.setSymbol("U+02716")' label="5. set to U+02716 &#10006;"/>
+                <button onClick='r2.setSymbol("z-icon-flag")' label="6. set to Font awesome z-icon-flag"/>
+                <button onClick='r2.setRating(4)' label="7. set rating to 4"/>
```
Comment (scribetw) [inline zktest/src/archive/test2/F86-ZK-3960.zul]: Add a case that the max = 5 but `setRating(10)`

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**